internal/userdirs/userdirs_test.go

```diff
@@ -0,0 +1,39 @@
+package userdirs_test
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/arm/remoteproc-runtime/internal/userdirs"
+	"github.com/stretchr/testify/require"
+)
+
+func TestRuntimeDir(t *testing.T) {
+	t.Run("when XDG_RUNTIME_DIR is set, returns its value", func(t *testing.T) {
+		testDir := "/tmp/xdg_runtime_test"
+		t.Setenv("XDG_RUNTIME_DIR", testDir)
+		want := filepath.Join(testDir, ".remoteproc-runtime")
+		got, err := userdirs.RuntimeDir()
+		require.NoError(t, err)
+		require.Equal(t, want, got)
```

Comment:
```suggestion
		testDir := "/tmp/xdg_runtime_test"
		t.Setenv("XDG_RUNTIME_DIR", testDir)
		
		
		got, err := userdirs.RuntimeDir()
		
		require.NoError(t, err)
		want := filepath.Join(testDir, ".remoteproc-runtime")
		require.Equal(t, want, got)
```

NIT: It's nice if tests are formatted like this where feasible - [Arrange Act Assert](https://automationpanda.com/2020/07/07/arrange-act-assert-a-pattern-for-writing-good-tests/).